tests/unit/test_download.py

```diff
@@ -413,6 +413,50 @@ def test_unpack_file_url_thats_a_dir(self, tmpdir, data):
         assert os.path.isdir(os.path.join(self.build_dir, 'fspkg'))
 
 
+@pytest.mark.parametrize('exclude_dir', [
+    '.nox',
+    '.tox'
+])
+def test_unpack_file_url_with_excluded_dirs(exclude_dir):
+
+    def touch(path):
+        with open(path, 'a'):
+            os.utime(path, None)
+
+    src_dir = mkdtemp()
```

Comment:
We can use the [`tmpdir`](https://github.com/pypa/pip/blob/82dbcdae87f7a8aaaf1a9bae663a5fac8c856a40/tests/conftest.py#L78) pytest fixture and make `src_dir` and `dst_dir` subdirectories of it, then they will be cleaned up automatically at the end of the test. So like:

1. add `tmpdir` as an argument to this function
2. `src_dir = tmpdir.joinpath('src')` or `src_dir = tmpdir / 'src'`